### PR TITLE
[go-gen] Add generation of parsing time attributes in requests

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of parsing blocks for attributes of type `date`, `time` or `datetime`
  * Request struct now makes these a string type, and are parsed before passing into DAO input struct
  * Cannot do inline because of error handling
* Optimises imports
* Updates tests

This bug was discovered with `temple test`, @jaylees14 tested this fix.

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala [hunk 1]
```diff
@@ -226,6 +226,64 @@ object GoServiceMainHandlersGenerator {
...
 
+  private def generateParseTimeBlock(name: String, attributeType: AttributeType): String =
+    mkCode.lines(
+      genDeclareAndAssign(
+        genMethodCall(
+          "time",
+          "Parse",
+          attributeType match {
+            case DateType     => doubleQuote("2006-01-02")
+            case TimeType     => doubleQuote("15:04:05.999999999")
```
Comment (Lewiky) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala:237]: What do these hard coded values mean?

Comment (jaylees14) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala:237]: They're layouts according to the reference time of 
`January 2nd at 15:04:05 2006 in timezone +07:00` (1-2-3-4-5-6-7)

https://gobyexample.com/time-formatting-parsing

**Review by jaylees14 — APPROVED** [review]
LGTM

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala:237]: Dumb right

**Review by Lewiky — APPROVED** [review]
sure

**Pull request merged**